assets/RNAseq_report.Rmd

```diff
@@ -1780,6 +1785,180 @@ pg[['x']][['layout']][['annotations']][[1]][['y']] <- -0.05
+
+    # convert zeros to small value (as in EnhancedVolcano)
+    DE_genes[which(DE_genes[['padj']] == 0), 'padj'] <- min(
+    DE_genes[which(DE_genes[['padj']] != 0), 'padj'], na.rm = TRUE) * 10^-1
+
+    # Get Number of DE features
+    numDE <- DE_genes %>%
+        filter(padj <= params$adj_pval_threshold, abs(log2FoldChange) >= params$logFC_threshold) %>%
+        nrow()
+
+    # Plot old volcano version static and save
+    DE_genes$logpval <- -log10(DE_genes$padj)
+    # Save true value for hovertext
+    DE_genes$truelogpval <- DE_genes$logpval
+    DE_genes <- DE_genes %>% mutate(color = ifelse((abs(DE_genes$log2FoldChange) >= params$logFC_threshold) &
+                        (DE_genes$logpval >= -log10(params$adj_pval_threshold)),
+                        "red", "black"))
+    log2FoldChange_min <- min(DE_genes$log2FoldChange)
+    log2FoldChange_max <- max(DE_genes$log2FoldChange)
+    log2FoldChange_quotient <- length(seq(log2FoldChange_min, log2FoldChange_max, 10))
+
+    pg <- ggplot(DE_genes, aes(x=log2FoldChange, y=logpval, text=paste("Gene: ", gene_name,
+                                                                        "<br>", "Log2FC: ", formatC(log2FoldChange, digits=2),
+                                                                        "<br>", "-Log10padj: ", formatC(truelogpval, digits=2, format='f')))) +
+            geom_point(alpha=0.5, aes(colour=ifelse(color == "red", "DE features",
+                                                                    "Non_DE features")),
+                                                                    na.rm=TRUE) +
+            geom_hline(yintercept = -log10(params$adj_pval_threshold), linewidth = 0.2, color = "grey") +
+            geom_vline(xintercept = -1, linewidth = 0.2, color = "grey") +
+            geom_vline(xintercept = 1, linewidth = 0.2, color = "grey") +
+            scale_color_manual(name = "", values=c("red", "black")) +
+            scale_y_continuous(limits = c(-1, max(-log10(DE_genes[['padj']]), na.rm=TRUE) + 5)) +
+            ylab("-log 10 padj") + xlab("Log2 Fold Change") +
+            theme(panel.spacing = unit(ifelse(length(allgenes_files)>2, 0.1, 0.6), "lines")) +
+            scale_x_continuous(breaks = seq(-200, 200, ifelse(log2FoldChange_quotient<3, 1, 5))) +
+            ggtitle(paste("DE results", DE_genes$contrast[1]),
+                subtitle=paste0("total=", thousandFormat(nrow(DE_genes)), ", DE_features=", thousandFormat(numDE)))
+
+
+    # Save static plot
+    ggsave(filename = file.path(wd, "/differential_gene_expression", "plots", "volcano_plots", paste0(fname, ".png")),
+    plot = pg,
+    width = 8,
+    height = 5)
+
+    # Define smaller plotly size
+    ppg <- partial_bundle(ggplotly(pg, tooltip = "text")) %>% layout(autosize = F, width = 600, height = 400)
+
+    # Export interactive plot
+    saveWidget(ppg, file = file.path(wd, "/differential_gene_expression", "plots", "volcano_plots", paste0(fname, ".html")))
```

Comment:
I am surprised by the `"/differential_gene_expression"`, why the slash?

Reply:
Yes, that is counterintuitive but it follows the "saving strategy" of files in the `RNAseq_report.Rmd` format.

This could all be refactored, but I decided to adhere to the rest of the code.